src/kernels/kernelRefreshIndicator.node.unit.test.ts

```diff
@@ -0,0 +1,230 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import * as fakeTimers from '@sinonjs/fake-timers';
+import { instance, mock, reset, verify, when } from 'ts-mockito';
+import { Disposable, EventEmitter, NotebookControllerDetectionTask } from 'vscode';
+import { disposeAllDisposables } from '../platform/common/helpers';
+import { IDisposable } from '../platform/common/types';
+import { KernelRefreshIndicator } from './kernelRefreshIndicator.node';
+import { IKernelFinder } from './types';
+import { mockedVSCodeNamespaces } from '../test/vscode-mock';
+import { IPythonExtensionChecker } from '../platform/api/types';
+import { IInterpreterService } from '../platform/interpreter/contracts';
+import { createDeferred } from '../platform/common/utils/async';
+import { InteractiveWindowView, JupyterNotebookView } from '../platform/common/constants';
+
+suite('Kernel Refresh Indicator (node)', () => {
```

Comment:
Should this live in test/kernels?

Reply:
no, withbnew code we try to keep unit tests closev to the corresponding source file.